packages/walletconnect-v2/src/index.ts

```diff
@@ -86,27 +102,54 @@ export class WalletConnect extends Connector {
     this.events.emit(URI_AVAILABLE, uri)
   }
 
+  private async initializeProvider(
+    desiredChainId: number | undefined = this.defaultChainId
+  ): Promise<WalletConnectProvider> {
+    const ethProviderModule = await import('@walletconnect/ethereum-provider')
+    const rpcMap = this.rpcMap ? await getBestUrlMap(this.rpcMap, this.timeout) : undefined
+    const chainProps = this.getChainProps(desiredChainId)
+
+    this.provider = await ethProviderModule.default.init({
+      ...this.options,
+      ...chainProps,
+      rpcMap,
+    })
+
+    return this.handleProviderEvents(this.provider)
+  }
+
+  // Helper function to reorder chains array based on desiredChainId
+  private reorderChainsBasedOnId(
+    chains: number[] | undefined,
+    desiredChainId: number | undefined
+  ): number[] | undefined {
+    if (!chains || !desiredChainId || !chains.includes(desiredChainId) || chains.length === 0) {
+      return chains
+    }
+    return getChainsWithDefault(chains, desiredChainId)
+  }
+
+  private getChainProps(desiredChainId: number | undefined = this.defaultChainId): ChainsProps {
+    // Reorder chains and optionalChains if necessary
+    const orderedChains = this.reorderChainsBasedOnId(this.options.chains, desiredChainId)
+    const orderedOptionalChains = this.reorderChainsBasedOnId(this.options.optionalChains, desiredChainId)
+
+    // Validate and return the result
+    if (isArrayOneOrMore(orderedChains)) {
```

Comment:
Why is this function necessary? Can't we just do `orderedChains.length > 0`?

Reply:
unfortunately no. This is a type guard that explicitly returns the named type that WC requires. If we did `orderedChains.length > 0` we would still get `number[]` which TS interprets as a variable length array and therefore not compliant with the WC requirement that at least one of the arrays is length > 0

Reply:
Makes sense!